docs/developers/content-creation/creating-a-role.md

```diff
@@ -40,3 +44,526 @@ suicideMultiplier = -1
 ```
 
 [[Find them in the role base]](https://github.com/TTT-2/TTT2/blob/master/lua/terrortown/entities/roles/ttt_role_base/shared.lua)
+
+## Setting up the files 
+
+Before we can dive into the lua programming, we must set up the role’s file structure, so that the role is detected by TTT2 as such. But this is not only depending on the file structure. Another important aspect is the initial code, which is the same for every role.
+
+### Folder structure
+
+Every role has the same folder structure. As a little side note: You can name the role’s folder like you want, but if you want to fit the most other roles, you should name the folder  `ttt2-role_[abbreviation of the role]”`.
+
+Inside this folder you will need another folder that must be called “lua” and then “terrortown”. In the terrortown-folder the structure is split into 3 parts:
+1. the files for the translation
+2. the main code
+3. the convar files 
+
+#### Translation files
+
+For the translation files, you must fit the the following structure: `tt2-role_test/lua/terrortown/lang/en/tester.lua”`. 
+
+Let me explain this structure, before we’re getting to the next one. As I explained above `ttt2-role_test` is the parent folder. After that we’ve got the lua and terrortown folders. These are needed to tell TTT2 that there are files, that it shall load. The “lang”-folder contains all languages that this role is translated to. Since the global language is English and there is no getting around this language in programming, you should definitely offer your role in English. If you want to add other translations you just can add here other language folders. For example:
+
+`ttt2-role_test/lua/terrortown/lang/de/tester.lua`
+
+This would also add a german translation. Or:
+
+`ttt2-role_test/lua/terrortown/lang/it/tester.lua`
+
+For an italian translation. 
```

Comment:
if you show it in such detail, mention that the lang identifier in the path is not important for the language itself, but only for the readability. The folder could have any name